config/test_example.toml

```diff
@@ -0,0 +1,12 @@
+[subnets]
+
+[subnets.root]
+path = "/root"
+rpc_api = "https://example.org/rpc/v0"
```

Comment:
shouldn't we call it `http_url` to be specific? Since the `JsonRpcClient` uses `ws`, if I pass in `ws`, will it work?

Reply:
That's a really good point. The current `JsonRpcClient` implementation uses HTTP for the `request` method and WS for the `subscribe` method. This is not ideal, particularly because each instance is associated with a specific URL. This means that each instance of the current client implementation can only actually call one of the two methods of the trait. This is far from ideal.

Here's what I suggest as a solution:
- The config file should have a field for HTTP and another for WS (e.g., `jsonrpc_api_http` and `jsonrpc_api_ws`. (I'll do this in the PR)
- Each implementation of the `JsonRpcClient` should use a single transport protocol. Instead of the current `JsonRpcClientImpl`, we should have a `HttpJsonRpcClient` and a `WsJsonRpcClient` implementation, both implementing the `JsonRpcClient` trait. (I'll do this in a follow-up PR)